charts/cryostat/templates/tests/test-core-connection.yaml

```diff
@@ -12,13 +12,11 @@ spec:
       image: registry.access.redhat.com/ubi8/ubi:latest
       command:
         - '/bin/bash'
-        - '-ec'
+        - '-exc'
         - >
           dnf install --disableplugin=subscription-manager -yq jq;
           curl -sSf --retry 10 --retry-connrefused -o /tmp/out.json http://{{ include "cryostat.fullname" . }}:{{ .Values.core.service.httpPort }}/health;
           cat /tmp/out.json;
-          jq -e '{{ printf "(.cryostatVersion | test(\"^v%s\")) and .datasourceAvailable == %t" (.Chart.AppVersion | squote) (not .Values.minimal) }}' /tmp/out.json;
-          {{- if not .Values.minimal }}
-          curl -sSf --retry 10 --retry-connrefused http://{{ include "cryostat.fullname" . }}-grafana:{{ .Values.grafana.service.port }}/api/health
-          {{- end }}
+          jq -e '{{ printf "(.cryostatVersion | test(\"^v%s(-snapshot)?\"))" (.Chart.AppVersion | trimSuffix "-dev" | squote) }}' /tmp/out.json;
```

Comment:
I'm worried this will miss a case where a Helm release attempts to mistakenly ship with a snapshot version. What about adding a branch that requires an exact match for releases like:
```
{{- if hasSuffix "-dev" .Chart.AppVersion }}
jq -e '{{ printf "(.cryostatVersion | test(\"^v%s\"))" (.Chart.AppVersion | squote) }}' /tmp/out.json;
{{- else }}
...
```